src/hypothesis/strategies.py

```diff
@@ -982,32 +987,61 @@ def fractions(min_value=None, max_value=None, max_denominator=None):
 
     If max_denominator is not None then the absolute value of the denominator
     of any generated values is no greater than max_denominator. Note that
-    max_denominator must be at least 1.
+    max_denominator must be None or a positive integer, and the denominator
```

Comment:
I don't believe that you have demonstrated that the current version is broken in any fundamental way. It is probably easy to return an error in the case where the arguments specified can not yield any values, and that would probably be a good idea.